src/error.rs

```diff
@@ -110,14 +148,11 @@ impl Error {
         Error { kind: kind, msg: msg }
     }
     
-    /// Get the error kind
-    pub fn kind(&self) -> ErrorKind {
-        self.kind
-    }
-    
-    /// Get the error message
-    pub fn msg(&self) -> &'static str {
-        self.msg
+    /// Take the cause, if any. This allows the embedded cause to be extracted.
+    /// This uses `Option::take`, leaving `self` with no cause.
+    #[cfg(feature="std")]
+    pub fn take_cause(&mut self) -> Option<Box<stdError + Send + Sync>> {
```

Comment:
`stdError` does not imply `Clone` and I didn't wish to add that, so this (or `unwrap_cause`) would be the only way to get the cause out, which I think might sometimes be useful.

I did consider just making `cause` a public field (with a hidden dummy field to prevent construction) but wasn't so sure about that.